Tests/Client/Unit Tests/Client+DevicesTests.swift

```diff
@@ -198,4 +198,59 @@ extension Client_DevicesTests {
             body: nil
         )
     }
+
+    func test_removeDevice_handlesSuccess() throws {
+        // Setup
+        let device = Device("device_id_\(UUID().uuidString)")
+        var user = client.user
+        user.devices = [device]
+        user.currentDevice = device
+        client.set(user: user, token: "test_token")
+
+        assert(user.devices == [device])
```

Comment:
Why not `XCTAssertEqual`?

Reply:
Because this is not what this test tests. I'm just checking the starting conditions are correct. If this assert fails other asserts in this tests are not valid.

Reply:
Hmm I wasn't aware `assert` and XCTAssert` were behaving differently (one stops and one marks), TIL 👍 
So if we have some logical failure to cause these `assert`s to fail, how will our CI respond?

Reply:
The behavior will be similar, it fails. 
I like doing this to make clear what the test actually tests. `assert` is to verify certain conditions unrelated to the functionality under test, `XCTAssert....` verify the results of the action under tests.
It's not something widely used. I do it to increase the readability of the tests.
I'm OK with changing to XCTAssert if it confuses you.

Reply:
Now that I know the reasoning, I'm ok with keeping it. Probably I wouldn't use this approach if I were to write tests, since these `assert`s are testing the underlying structs, which feels unnecessary to me.
You should write a testing document for us, explaining our DSL and all :D